uxarray/grid.py

```diff
@@ -73,9 +73,21 @@ def __init__(self, dataset, **kwargs):
 
         # check if initializing from verts:
         if isinstance(dataset, (list, tuple, np.ndarray)):
-            self.vertices = dataset
-            self.__from_vert__()
-            self.source_grid = "From vertices"
+            dataset = np.asarray(dataset)
+            # grid with multiple faces
+            if dataset.ndim == 3:
+                self.__from_vert__(dataset)
+                self.source_grid = "From vertices"
+            # grid with a single face
+            elif dataset.ndim == 2:
+                dataset = np.array([dataset])
+                self.__from_vert__(dataset)
+                self.source_grid = "From vertices"
+            else:
+                raise RuntimeError(
+                    "Input vertices don't have correct dimensions (Expected dimension will be 3: [nMesh2_face, "
+                    "nMesh2_node, Two/Three]). Or dimesion of 2 when only one face is passed"
```

Comment:
Make sure to run pre-commit on this, since inline code changes through Github can sometimes break with formatting.

```suggestion
                raise RuntimeError(
                    f"Invalid Input Dimension: {dataset.ndim}. Expected dimension should be "
                    f"3: [nMesh2_face, nMesh2_node, Two/Three] or 2 when only one "
                    f"one face is passed in."
```